dlt/destinations/impl/dremio/dremio.py

```diff
@@ -0,0 +1,251 @@
+    ) -> List[str]:
+        sql = super()._get_table_update_sql(table_name, new_columns, generate_alter)
+
+        if not generate_alter:
+            partition_list = [
+                self.capabilities.escape_identifier(c["name"])
+                for c in new_columns
+                if c.get("partition")
+            ]
+            if partition_list:
+                sql[0] += "\nPARTITION BY (" + ",".join(partition_list) + ")"
+
+            sort_list = [
+                self.capabilities.escape_identifier(c["name"]) for c in new_columns if c.get("sort")
+            ]
+            if sort_list:
+                sql[0] += "\nLOCALSORT BY (" + ",".join(sort_list) + ")"
+
+        return sql
+
+    def _from_db_type(
+        self, bq_t: str, precision: Optional[int], scale: Optional[int]
+    ) -> TColumnType:
+        return self.type_mapper.from_db_type(bq_t, precision, scale)
+
+    def _get_column_def_sql(self, c: TColumnSchema, table_format: TTableFormat = None) -> str:
+        name = self.capabilities.escape_identifier(c["name"])
+        return (
+            f"{name} {self.type_mapper.to_db_type(c)} {self._gen_not_null(c.get('nullable', True))}"
+        )
+
+    def get_storage_table(self, table_name: str) -> Tuple[bool, TTableSchemaColumns]:
+        def _null_to_bool(v: str) -> bool:
+            if v == "NO":
+                return False
+            elif v == "YES":
+                return True
+            raise ValueError(v)
+
+        fields = self._get_storage_table_query_columns()
+        table_schema = self.sql_client.fully_qualified_dataset_name(escape=False)
+        db_params = (table_schema, table_name)
+        query = f"""
+SELECT {",".join(fields)}
+    FROM INFORMATION_SCHEMA.COLUMNS
+WHERE
+    table_catalog = 'DREMIO' AND table_schema = %s AND table_name = %s ORDER BY ordinal_position;
+"""
+        rows = self.sql_client.execute_sql(query, *db_params)
+
+        # if no rows we assume that table does not exist
+        schema_table: TTableSchemaColumns = {}
+        if len(rows) == 0:
+            return False, schema_table
+        for c in rows:
+            numeric_precision = (
+                c[3] if self.capabilities.schema_supports_numeric_precision else None
+            )
+            numeric_scale = c[4] if self.capabilities.schema_supports_numeric_precision else None
+            schema_c: TColumnSchemaBase = {
+                "name": c[0],
+                "nullable": _null_to_bool(c[2]),
+                **self._from_db_type(c[1], numeric_precision, numeric_scale),
+            }
+            schema_table[c[0]] = schema_c  # type: ignore
+        return True, schema_table
+
+    def _create_merge_followup_jobs(self, table_chain: Sequence[TTableSchema]) -> List[NewLoadJob]:
+        def _new_temp_table_name(name_prefix: str) -> str:
+            return self.sql_client.make_qualified_table_name(f"_temp_{name_prefix}_{uniq_id()}")
+
+        class DremioMergeJob(SqlMergeJob):
```

Comment:
don't define this as an inner class, you can put it at the top.

Reply:
are you sure there won't be any name colission if the temp tables are all named the same? why could you not use the uniq_id of the standard merge job?

Reply:
ah, ok, you need to make a qualified table name right? why not ammend the main sqlmergejob to do this? i hope it would work for all destination.. :)

Reply:
Yeah this is a bit wacky. Dremio doesn't have `CREATE TEMP TABLE`, so I needed to create a normal table in the same location. I'm not sure I can easily move this logic into `SqlMergeJob` the way everything is currently written using `@classmethod`s.

Reply:
Ok, I added the sql_client as an arg for the classmethod and now it looks a bit nicer :)